src/diffusion_for_multi_scale_molecular_dynamics/models/axl_diffusion_lightning_model.py

```diff
@@ -180,73 +254,164 @@ def _generic_step(
         assert (
             RELATIVE_COORDINATES in batch
         ), f"The field '{RELATIVE_COORDINATES}' is missing from the input."
+
+        assert (
+            ATOM_TYPES in batch
+        ), f"The field '{ATOM_TYPES}' is missing from the input."
+
         x0 = batch[RELATIVE_COORDINATES]
         shape = x0.shape
         assert len(shape) == 3, (
             f"the shape of the RELATIVE_COORDINATES array should be [batch_size, number_of_atoms, spatial_dimensions]. "
             f"Got shape = {shape}."
         )
+
+        a0 = batch[ATOM_TYPES]
         batch_size = self._get_batch_size(batch)
+        atom_shape = a0.shape
+        assert len(atom_shape) == 2, (
+            f"the shape of the ATOM_TYPES array should be [batch_size, number_of_atoms]. "
+            f"Got shape = {atom_shape}"
+        )
+
+        lvec0 = batch[
```

Comment:
Why not call this `l0` instead of `lvec0`? We use `x0` and `a0`, I think that it will be clear in this context.